extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -116,14 +124,14 @@ private void registerNativeImageResources(BuildProducer<NativeImageResourceBuild
         resource.produce(new NativeImageResourceBuildItem(nativeResources.toArray(new String[0])));
     }
 
-    private List<String> discoverApplicationMigrations(FlywayBuildConfig flywayBuildConfig)
+    private List<String> discoverApplicationMigrations(Collection<String> dataSourceNames)
             throws IOException, URISyntaxException {
         List<String> resources = new ArrayList<>();
         try {
-            List<String> locations = new ArrayList<>(flywayBuildConfig.locations);
-            if (locations.isEmpty()) {
-                locations.add("db/migration");
-            }
+            Set<String> locations = dataSourceNames.stream()
+                    .map(flywayBuildConfig::getConfigForDataSourceName)
+                    .flatMap(config -> config.locations.stream())
+                    .collect(Collectors.toSet());
```

Comment:
I would keep a list and stay ordered. Unordered things are bad :).

Reply:
ok

Reply:
ok